packages/mermaid/src/diagrams/xychart/parser/xychart.jison

```diff
@@ -0,0 +1,201 @@
+%lex
+%options case-insensitive
+
+%x string
+%x md_string
+%x title
+%x open_directive
+%x type_directive
+%x arg_directive
+%x close_directive
+%x acc_title
+%x acc_descr
+%x acc_descr_multiline
+%s axis_data
+%s data
+%s data_inner
+%%
+\%\%\{                                    { this.begin('open_directive'); return 'open_directive'; }
+<open_directive>((?:(?!\}\%\%)[^:.])*)    { this.begin('type_directive'); return 'type_directive'; }
+<type_directive>":"                       { this.popState(); this.begin('arg_directive'); return ':'; }
+<type_directive,arg_directive>\}\%\%      { this.popState(); this.popState(); return 'close_directive'; }
+<arg_directive>((?:(?!\}\%\%).|\n)*)      return 'arg_directive';
+\%\%(?!\{)[^\n]*                          /* skip comments */
+[^\}]\%\%[^\n]*                           /* skip comments */
+<axis_data>(\r?\n)                        { this.popState(); return 'NEWLINE'; }
+<data>(\r?\n)                             { this.popState(); return 'NEWLINE'; }
+[\n\r]+                                   return 'NEWLINE';
+\%\%[^\n]*                                /* do nothing */
+
+title                                     { this.begin("title"); return 'title'; }
+<title>(?!\n|;|#)*[^\n]*                  { this.popState(); return "title_value"; }
```

Comment:
I can't say I'm super familiar with regex application, but why are we using a negative lookahead at the beginning of the regex?

Reply:
changed with text